### DISCUSSION TITLE
Event Trigger 예제에서 생긴 문제

### DISCUSSION THREAD

**Discussion by earthboom** [naverz/zepeto-studio-kor#164] [5. Q&A]
안녕하세요.

ZEPETO 가이드에 있는 'Event Trigger'를 현 프로젝트에 그대로 적용해 보려고 했습니다.

마지막의 trigger.triggers.Add(Entry) 부분이 문제가 생기더군요.

Add부분의 함수가 없다고 하더군요.

해당 부분을 확인해보니, index.d.ts의 System.Collections.Genric 모듈의 List$1<T> 의 interface가 Add함수를 가지고 있지 않았습니다.

이 때문에, 이 문제를 어떻게 해결할 수 있는지 문의 드립니다.

감사합니다.

**Comment by hyeonjin-jo** [comment]
안녕하세요.

이 부분 현재 내용 파악되었으며 이후 수정 항목에 추가하겠습니다.

임시로 해결하는 방법은 아래 링크의 파일을 작업 중이신 프로젝트에 추가하시면 정상 동작합니다.

https://github.com/naverz/zepeto-script-sample/blob/main/Assets/1.%20Basic/7.%20EventTrigger/index.d.ts

감사합니다.

**Reply by smilejsu82** [reply]
안녕하세요 

월드 1.1.7 사용하고 있는데요 이거 동작 하나요?
![image](https://user-images.githubusercontent.com/50188795/163426904-c66a3129-3b81-4122-820e-64a9eba784a9.png)
`import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import { EventTrigger, EventTriggerType } from "UnityEngine.EventSystems";
import { Entry } from "UnityEngine.EventSystems.EventTrigger";

export default class EventTriggerSample extends ZepetoScriptBehaviour {

    Start() {
        var trigger = this.gameObject.GetComponent<EventTrigger>();

        // create eventsystem entry
        var entry = new Entry();

        // specify event type & delegation
        entry.eventID = EventTriggerType.PointerClick;
        entry.callback.AddListener(() => {
            console.log("click")
        });

        // register event entry
        trigger.triggers.Add(entry);
    }
}`
